internal/core/data/event_test.go

```diff
@@ -16,32 +16,185 @@ package data
 
 import (
 	"encoding/json"
+	"fmt"
 	"net/http"
 	"net/http/httptest"
 	"os"
 	"strconv"
 	"testing"
 
+	"github.com/edgexfoundry/edgex-go/internal/core/data/messaging"
+	"github.com/edgexfoundry/edgex-go/internal/pkg/db"
 	"github.com/edgexfoundry/edgex-go/internal/pkg/db/memory"
 	"github.com/edgexfoundry/edgex-go/pkg/clients/logging"
+	"github.com/edgexfoundry/edgex-go/pkg/clients/metadata/mocks"
 	"github.com/edgexfoundry/edgex-go/pkg/models"
 	"github.com/gorilla/mux"
+	"github.com/stretchr/testify/mock"
+	"gopkg.in/mgo.v2/bson"
+	"sync"
+	"time"
 )
 
 var testEvent models.Event
 var testRoutes *mux.Router
 
+const (
+	TEST_DEVICE_NAME string = "Test Device"
```

Comment:
Using all capitals is not Go idiomatic. I think that `testDeviceName` would work fine here.